app/cdap/components/AbstractWidget/index.tsx

```diff
@@ -59,29 +59,37 @@ export interface IWidgetProps<T = any> {
 }
 
 interface IAbstractWidgetProps extends IWidgetProps {
-  type: string;
+  type?: string;
```

Comment:
When is this optional?

Reply:
there were a ton of type errors on unit tests without this

Reply:
actually nvm, the errors were in stories
```
[3] ERROR in /Users/podesta/goog/cdap-ui/app/cdap/components/AbstractWidget/Widgets.stories.tsx(81,38):
[3] TS2741: Property 'value' is missing in type '{ type: string; }' but required in type 'IAbstractWidgetProps'.
[3] ERROR in /Users/podesta/goog/cdap-ui/app/cdap/components/AbstractWidget/Widgets.stories.tsx(83,27):
[3] TS2741: Property 'value' is missing in type '{ type: string; }' but required in type 'IAbstractWidgetProps'.
[3] ERROR in /Users/podesta/goog/cdap-ui/app/cdap/components/AbstractWidget/Widgets.stories.tsx(86,4):
[3] TS2741: Property 'value' is missing in type '{ type: string; widgetProps: { dropdownOptions: { value: string; label: string; }[]; }; }' but required in type 'IAbstractWidgetProps'
```

Reply:
whoops i got these mixed up haha - this one wasn't necessary, `values?` was

Reply:
`type?` is necessary because default widget props doesn't contain it.